src/FffGcodeWriter.cpp

```diff
@@ -514,6 +514,20 @@ void FffGcodeWriter::processRaft(const SliceDataStorage& storage)
         infill_comp.generate(raft_polygons, raftLines);
         gcode_layer.addLinesByOptimizer(raftLines, &gcode_layer.configs_storage.raft_base_config, SpaceFillType::Lines);
 
+        // When we use raft, we need to make sure that all used extruders for this print will get primed on the first raft layer,
+        // and then switch back to the original extruder.
+        std::vector<uint32_t> extruder_order = calculateLayerExtruderOrder(storage, extruder_nr, layer_nr);
+        bool has_extruders_primed = false;
+        for (uint32_t each_extruder_need_prime : extruder_order)
+        {
+            setExtruder_addPrime(storage, gcode_layer, layer_nr, each_extruder_need_prime);
+            has_extruders_primed = true;
+        }
+        if (has_extruders_primed)
+        {
+            gcode_layer.setExtruder(extruder_nr);
```

Comment:
This is the last part of the first layer. Without the change, it will just print the first layer with an extruder and go on. In case we have dual extrusion, we do an extra extruder switch, and I think it makes sense to make sure that it switches back to the original extruder if there is any extruder switches here.